File: src/main/java/org/example/Bot.java
package org.example;

import io.github.cdimascio.dotenv.Dotenv;
import net.dv8tion.jda.api.JDA;
import net.dv8tion.jda.api.JDABuilder;
import net.dv8tion.jda.api.entities.Activity;
import net.dv8tion.jda.api.requests.GatewayIntent;

public class Bot {
  private static Dotenv config;

  public static void main(String[] args) throws Exception {
    config = Dotenv.configure().load();
    String token = config.get("BOT_TOKEN");

    JDA api =
        JDABuilder.createDefault(token)
            .enableIntents(GatewayIntent.MESSAGE_CONTENT)
            .addEventListeners(new Listener())
            .setActivity(Activity.playing("Filtering Spam Messages"))
            .build();

    api.awaitReady();
  }

  public Dotenv getConfig() {
    return config;
  }
}

File: src/main/java/org/example/Bot.kt
package org.example

import net.dv8tion.jda.api.JDABuilder
import net.dv8tion.jda.api.entities.Activity
import net.dv8tion.jda.api.requests.GatewayIntent
import org.example.commands.CommandListener
import org.example.commands.ExclamListener
import org.example.commands.commandList

object Bot {
    @Throws(Exception::class)
    @JvmStatic
    fun main(args: Array<String>) {
        val api =
            JDABuilder.createDefault(token)
                .enableIntents(GatewayIntent.MESSAGE_CONTENT)
                .addEventListeners(ExclamListener(), CommandListener())
                .setActivity(Activity.playing("Filtering Spam Messages"))
                .build()

        api.updateCommands().addCommands(commandList()).queue()
        api.awaitReady()
    }

    private val token: String
        get() = System.getenv("BOT_TOKEN")
}
